lineapy/utils/lineabuiltins.py

```diff
@@ -258,6 +262,29 @@ def l_unpack_ex(
     return [*before_list, middle_list, *after_list]
 
 
+@register
+def l_import(
+    name: str, base_module: types.ModuleType = None
+) -> types.ModuleType:
+    """
+    Imports and returns a module. If the base_module is provided, the module
+    will be a submodule of the base.
+
+    Marks the `base_module` as modified if provided.
+    """
+    assert "." not in name
+    full_name = base_module.__name__ + "." + name if base_module else name
+    try:
+        value = importlib.import_module(full_name)
+    # wrap all exceptions happening during import into a single importerror
+    # this can potentially be changed to a custom error if it messes with
+    # imports inside blackboxes
```

Comment:
Can you elaborate on when things would be messed with? What other errors besides `ImportError` would we see?

Reply:
The main motivation to do this because import errors get stacked with additional frames that we wont catch if we caught them as general Exceptions (this is happening when we initialize a UserException in the executor's execute_call)

What we are saying here is that any type of error that happens during import is wrapped up in ImportError (this could be a new custom error too, we just reused importerror). We have one test case where during import we encounter a divided by zero error. To specifically catch errors during import (and escape the additional "<frozen importlib" frames), we catch all exceptions encountered during import_module and re-raise them as importError (or a custom error) with the original cause intact. This original exception later gets picked, the frames get adjusted correctly accounting for the additional frames and repackaged as userexception.

Not sure if all this needs to be in the doc. There wont be much of a change here.